ground/src/test/java/com/google/android/ground/repository/SurveyRepositoryTest.kt

```diff
@@ -16,46 +16,136 @@
 package com.google.android.ground.repository
 
 import com.google.android.ground.BaseHiltTest
+import com.google.android.ground.coroutines.DefaultDispatcher
 import com.google.android.ground.model.Survey
 import com.google.android.ground.persistence.local.LocalDataStore
 import com.google.android.ground.persistence.local.LocalDataStoreModule
+import com.google.android.ground.persistence.local.LocalValueStore
 import com.google.android.ground.persistence.local.room.RoomLocalDataStore
 import com.google.android.ground.persistence.local.stores.LocalSurveyStore
-import com.google.common.collect.ImmutableMap
+import com.google.android.ground.rx.Loadable
+import com.google.android.ground.rx.Loadable.LoadState.ERROR
+import com.google.common.truth.Truth.assertThat
+import com.sharedtest.FakeData.SURVEY
+import com.sharedtest.persistence.remote.FakeRemoteDataStore
 import dagger.hilt.android.testing.BindValue
 import dagger.hilt.android.testing.HiltAndroidTest
 import dagger.hilt.android.testing.UninstallModules
+import io.reactivex.Completable
 import io.reactivex.Maybe
 import java8.util.Optional
 import javax.inject.Inject
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.advanceUntilIdle
+import kotlinx.coroutines.test.runTest
+import org.junit.Before
 import org.junit.Test
 import org.junit.runner.RunWith
 import org.mockito.ArgumentMatchers.anyString
 import org.mockito.InjectMocks
 import org.mockito.Mock
-import org.mockito.Mockito
+import org.mockito.Mockito.*
+import org.mockito.kotlin.any
 import org.robolectric.RobolectricTestRunner
 
+@OptIn(ExperimentalCoroutinesApi::class)
 @HiltAndroidTest
 @UninstallModules(LocalDataStoreModule::class)
 @RunWith(RobolectricTestRunner::class)
 class SurveyRepositoryTest : BaseHiltTest() {
   @BindValue @InjectMocks var mockLocalDataStore: LocalDataStore = RoomLocalDataStore()
```

Comment:
Can this be cleaned up in favor of mockSurveyStore already present in this test?

Reply:
Good catch, thanks. Fixed!

Reply:
Test broke after this change... I'll take a look.

Reply:
Oops, my bad. Please feel free to leave a todo to unblock the PR

Reply:
Unfortunately making that change will require additional refactors, since `LocalSurveyStore` is injected into both `SurveyRepository` and `LocalDataStore`. I'll punt on this for now, thanks! @scolsen FYI.

Reply:
:+1: